sui-execution/latest/sui-move-natives/src/object_runtime/mod.rs

```diff
@@ -501,62 +505,22 @@ impl ObjectRuntimeState {
         // Check new owners from transfers, reports an error on cycles.
         // TODO can we have cycles in the new system?
         check_circular_ownership(transfers.iter().map(|(id, (owner, _, _))| (*id, *owner)))?;
-        // determine write kinds
-        let writes: LinkedHashMap<_, _> = transfers
+        let written_objects: LinkedHashMap<_, _> = transfers
             .into_iter()
             .map(|(id, (owner, type_, value))| {
-                let write_kind =
-                    if input_objects.contains_key(&id) || loaded_child_objects.contains_key(&id) {
-                        debug_assert!(!new_ids.contains_key(&id));
-                        WriteKind::Mutate
-                    } else if id == SUI_SYSTEM_STATE_OBJECT_ID || new_ids.contains_key(&id) {
-                        // SUI_SYSTEM_STATE_OBJECT_ID is only transferred during genesis
-                        // TODO find a way to insert this in the new_ids during genesis transactions
-                        WriteKind::Create
-                    } else {
-                        WriteKind::Unwrap
-                    };
-                (id, (write_kind, owner, type_, value))
-            })
-            .collect();
-        // determine delete kinds
-        let mut deletions: LinkedHashMap<_, _> = deleted_ids
-            .into_iter()
-            .map(|(id, ())| {
-                debug_assert!(!new_ids.contains_key(&id));
-                let delete_kind =
-                    if input_objects.contains_key(&id) || loaded_child_objects.contains_key(&id) {
-                        DeleteKind::Normal
-                    } else {
-                        DeleteKind::UnwrapThenDelete
-                    };
-                (id, delete_kind)
+                if let Some(loaded_child) = loaded_child_objects.get_mut(&id) {
+                    loaded_child.is_modified = true;
+                }
+                (id, (owner, type_, value))
             })
             .collect();
-        // remaining by value objects must be wrapped
-        let remaining_by_value_objects = by_value_inputs
-            .into_iter()
-            .filter(|id| {
-                !writes.contains_key(id)
-                    && !deletions.contains_key(id)
-                    && !external_transfers.contains(id)
-            })
-            .collect::<Vec<_>>();
-        for id in remaining_by_value_objects {
-            deletions.insert(id, DeleteKind::Wrap);
-        }
-        // children that weren't deleted or transferred must be wrapped
-        for id in wrapped_children {
-            deletions.insert(id, DeleteKind::Wrap);
```

Comment:
I would like to verify this functionality is now moved to `fn record_execution_results` in temporary store? This causes some complications for shared object deletion and breaks the check that wrapped objects can't be deleted, which @amnn @tnowacki and I discussed we wanted to keep out of the temporary store. Question for Xun, Why are we moving these checks to the temporary_store? And do others feel like because of this change, it better to do the wrapped object check there despite our earlier discussions?

Reply:
The current implementation of `record_execution_results` is temporary. It's there such that the same data generation can be compatible with both effects v1 and v2. They will all go away in a future version. The check of wrapped shared object will still happen inside execution. I left a place specifically for checking wrapped shared object.

Reply:
This change when refactored on my branch breaks wrapping, passing as vectors, and several other tests in shared object deletion. What about the refactors to this function are needed to work with v2 so that when I attempt to fix this for sod, I don't break what you were aiming for here?

Reply:
I left a case of shared object wrapping marked as unreachable. So if you make it report an error it should work. But do let me know if there is some other case that I didn't think of.

Reply:
The case where you marked unreachable and commented as represents a wrapped shared object can also include a vector containing a shared object passed by value as well, and there is no way with the data maintained in your refactored version to tell that it is a vector, so unfortunately its not as simple as returning an error instead of unreachable

Reply:
Interesting. Wouldn't a vector of objects get flattened in the `inputs` given that it's a `Vec<InputValue>`?